### PR TITLE
add tooltip in base component

### PR DESCRIPTION
**Pull request by patelvarun**

### CODE REVIEW

File: packages/base/components/Tooltip/index.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { renderToStaticMarkup } from 'react-dom/server';
+import ReactTooltip from 'react-tooltip';
+import Box from '../Box';
+import ReactTooltipStyled from './ReactTooltipStyled';
+
+const getTags = string =>
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: I'm hesitant to have custom content styles/structure embedded in a shared tooltip component. If we add custom styles for every kind of content a tooltip can have this component will get very bloated. Instead, let's keep this component dumb, and have a local component in barrel that adds tag styles to this basic shared tooltip.

File: packages/base/components/Tooltip/ReactTooltipStyled.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+const ReactTooltipStyled = styled(ReactTooltip)`
+  max-width: ${p => p.maxWidth};
+  max-height: ${p => p.maxHeight};
+  display: flex;
+  flex-direction: column;
+  pointer-events: ${p => p.pointer};
+  .graphTooltip {
+    display: flex;
+    text-align: center;
+    font-size: ${p => p.theme.fontSizes.md};
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/ReactTooltipStyled.js]: We should use array scales here.

File: packages/base/components/Tooltip/ReactTooltipStyled.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    > * {
+      display: inline-block;
+      padding: 0 2px;
+    }
+    .highlight,
+    .title {
+      font-weight: ${p => p.theme.fontWeights.bold};
+    }
+    .subtitle {
+      color: ${p => p.theme.colors.gray4};
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/ReactTooltipStyled.js]: We need to use `colors.grays.4` here.

File: packages/base/components/Tooltip/ReactTooltipStyled.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      font-size: ${p => p.theme.fontSizes.sm};
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/ReactTooltipStyled.js]: We should use array scales here.

File: packages/base/components/Tooltip/ReactTooltipStyled.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      width: 4px;
+    }
+
+    &::-webkit-scrollbar-track {
+      box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
+    }
+
+    &::-webkit-scrollbar-thumb {
+      background: white;
+      border-radius: ${p => p.theme.radii.sm};
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/ReactTooltipStyled.js]: We should use array scales here.

File: packages/base/components/Tooltip/ReactTooltipStyled.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    }
+  }
+
+  .tag {
+    ${p => css`
+      border: 1px solid ${p.theme.colors.gray5};
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/ReactTooltipStyled.js]: We should use `colors.grays.5` here.

File: packages/base/components/Tooltip/ReactTooltipStyled.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      border-radius: ${p.theme.radii.md};
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/ReactTooltipStyled.js]: We should use array scales here.

File: packages/base/components/Tooltip/ReactTooltipStyled.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+      display: inline-block;
+      flex: auto;
+      margin: 3px;
+      padding: ${p.theme.space.xxxs} ${p.theme.space.xs};
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/ReactTooltipStyled.js]: We should use array scales here.

File: packages/base/components/Tooltip/index.js (hunk 1 above)
Comment (patelvarun) [reply packages/base/components/Tooltip/index.js]: @cadejscroggins  - Yes, we can move tag style in our barrel local component. But we can't apply theme props (like theme.colors, etc..) to the styles since the tooltip content HTML in string format.  Let's discuss on call.

File: packages/base/components/Tooltip/index.js [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+      ...rest
+    } = this.props;
+
+    const randomId = Math.random()
+      .toString(36)
+      .substring(7);
+
+    return (
+      <>
+        {showDataTip && (
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: Instead of checking `showDataTip` let's just check to see if `children` exists.

File: packages/base/components/Tooltip/index.js [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+          >
+            {children}
+          </Box>
+        )}
+        <Box
+          as={ReactTooltip}
+          id={isMultiple ? id : `tooltip_${randomId}`}
+          maxHeight={maxHeight}
+          maxWidth={maxWidth}
+          place={place}
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: These don't need to be explicitly passed.

File: packages/base/components/Tooltip/index.js [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+      isMultiple,
+      maxWidth,
+      maxHeight,
+      place,
+      sx,
+      themeType,
+      ...rest
+    } = this.props;
+
+    const randomId = Math.random()
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: Let's require an id to be passed. I don't want to rely on chance for unique ids.

File: packages/base/components/Tooltip/index.js [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+          </Box>
+        )}
+        <Box
+          as={ReactTooltip}
+          id={isMultiple ? id : `tooltip_${randomId}`}
+          maxHeight={maxHeight}
+          maxWidth={maxWidth}
+          place={place}
+          sx={{
+            '.tooltipContentWrapper': {
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: If possible, our default styling should not apply any class names. Can we apply these styles to the root of `sx`, with `!important`, if necessary? This makes it easier to override the styles in the app.

File: packages/base/components/Tooltip/index.js [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+              '&::-webkit-scrollbar': {
+                width: '4px',
+              },
+              '&::-webkit-scrollbar-thumb': {
+                background: p => p.colors.white,
+                borderRadius: p => p.radii[0],
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: We don't need to use functions when `!important` isn't used.

File: packages/base/components/Tooltip/index.js [hunk 14]
```diff
@@ -0,0 +1,112 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactTooltip from 'react-tooltip';
+import Box from '../Box';
+
+class Tooltip extends React.Component {
+  componentDidUpdate() {
+    ReactTooltip.rebuild();
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: Why is this necessary? If possible, we should make this a functional component.

File: packages/base/components/Tooltip/index.js [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  render() {
+    const {
+      children,
+      content,
+      id,
+      disable,
+      showDataTip,
+      isMultiple,
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: This prop isn't self explanatory at all. We should remove this and pass custom `pointerEvents` styles where necessary.

File: packages/base/components/Tooltip/index.js (hunk 14 above)
Comment (patelvarun) [reply packages/base/components/Tooltip/index.js]: We need this for the graph dashboard to initialize tooltip after data loaded.

File: packages/base/components/Tooltip/index.js [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+            {...rest}
+          >
+            {children}
+          </Box>
+        )}
+        <Box
+          as={ReactTooltip}
+          id={id}
+          maxHeight={maxHeight}
+          maxWidth={maxWidth}
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: These still need to be removed.

File: packages/base/components/Tooltip/index.js [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+      </>
+    );
+  }
+}
+
+Tooltip.propTypes = {
+  children: PropTypes.node,
+  content: PropTypes.node,
+  disable: PropTypes.bool,
+  id: PropTypes.string,
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: This should be required/not have a default value.

File: packages/base/components/Tooltip/index.js [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+  maxWidth: PropTypes.string,
+  place: PropTypes.string,
+  sx: PropTypes.shape({}),
+  type: PropTypes.string,
+};
+
+Tooltip.defaultProps = {
+  children: null,
+  content: '',
+  disable: false,
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: This is the default and can be removed.

File: packages/base/components/Tooltip/index.js [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+  id: 'graph_tooltip',
+  maxHeight: '202px',
+  maxWidth: '60ch',
+  place: 'top',
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: This is the default and can be removed.

File: packages/base/components/Tooltip/index.js [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+  sx: {},
+  type: 'dark',
```
Comment (cadejscroggins) [inline packages/base/components/Tooltip/index.js]: This is the default and can be removed.
